src/BuildScriptGeneratorCli/Commands/BuildCommand.cs

```diff
@@ -129,6 +129,9 @@ internal override int Execute(IServiceProvider serviceProvider, IConsole console
             var oryxVersion = Program.GetVersion();
             var oryxCommitId = Program.GetMetadataValue(Program.GitCommit);
             var oryxReleaseTagName = Program.GetMetadataValue(Program.ReleaseTagName);
+
+            var options = serviceProvider.GetRequiredService<IOptions<BuildScriptGeneratorOptions>>().Value;
```

Comment:
This change is only going to be available for the `oryx build` command -- we want to make sure that the `USER_CALLER_ID` environment variable is read for all of our commands in the case that we want to check the frequency by which any partner teams are calling our commands. We should find a way to generalize this change you've added such that it's readily available for all of our commands (my vote is to find a way to put it in the `CommandsBase` class).

Reply:
thanks for the suggestion, Cormac. When I discussed with Dan regarding the potential reports we can generate using this telemetry, the main focus was to get telemetry in the `build` command. Hence, I added it in the `build` command.
your suggestion is also a nice one. Let me find a way to do it.

Reply:
@cormacpayne In the `CommandBase` class we have an event telemetry `logger?.LogTimedEvent(this.GetType().Name))`. I have used this event and added the `callerId` here because this event is created whenever a `*Command` class object is executed (example: BuildCommand, RunScriptCommand, PrepareEnvironmentCommand, etc).  
However, for other command the usage is quite low compared to the `BuildCommand`. Here is a quick stat I got from the AppInsight (in last 24 hours):
count of usage of `BuildCommand`:  30,476
count of usage of other Commands: 8

Based on that data I am just wondering do we really need to consider other `commands`?

Reply:
We do want to consider other Oryx CLI commands for two reasons: (1) we just collaborated with the Container App folks to integrate a call to `oryx dockerfile` into their `az containerapp up` command that's part of the Azure CLI, so we want to make sure that we're recording how often that command is being called by the Azure CLI, and (2) the telemetry is already in place in the `CommandBase` class to allow us to track this information across all of our commands, so it's better to generalize our telemetry collection where we can rather than have it collected for only a single command.

Reply:
makes sense. I already added it inside `CommandBase `.